src/test/java/seedu/address/logic/commands/ListStudentCommandTest.java

```diff
@@ -0,0 +1,521 @@
+        ListStudentCommand fourthCommand = new ListStudentCommand(null, VALID_TUTORIAL_NAME_2);
+        ListStudentCommand allStudentCommand = new ListStudentCommand(null, null);
+
+        // same object -> returns true
+        assertTrue(firstCommand.equals(firstCommand));
+        assertTrue(thirdCommand.equals(thirdCommand));
+        assertTrue(allStudentCommand.equals(allStudentCommand));
+
+        // same values -> returns true
+        ListStudentCommand firstCommandCopy = new ListStudentCommand(INDEX_ONE, null);
+        assertTrue(firstCommand.equals(firstCommandCopy));
+        ListStudentCommand thirdCommandCopy = new ListStudentCommand(null, VALID_TUTORIAL_NAME_1);
+        assertTrue(thirdCommand.equals(thirdCommandCopy));
+        ListStudentCommand allStudentCommandCopy = new ListStudentCommand(null, null);
+        assertTrue(allStudentCommand.equals(allStudentCommandCopy));
+
+        // different types -> returns false
+        assertFalse(firstCommand.equals(1));
+
+        // null -> returns false
+        assertFalse(firstCommand.equals(null));
+
+        // different values -> returns false
+        assertFalse(firstCommand.equals(secondCommand));
+        assertFalse(thirdCommand.equals(fourthCommand));
+        assertFalse(firstCommand.equals(thirdCommand));
+    }
+
+    @Test
+    public void execute_noIndexTutNameInput_success() {
+        String expectedMessage = ListStudentCommand.MESSAGE_SUCCESS;
+        ListStudentCommand command = new ListStudentCommand(null, null);
+        expectedModel.updateFilteredStudentList(expectedModel.PREDICATE_SHOW_ALL_STUDENTS);
+        assertCommandSuccess(command, model, expectedMessage, expectedModel);
+    }
+
+    @Test
+    public void execute_indexInput_success() throws Exception {
+        String expectedMessage = String.format(ListStudentCommand.MESSAGE_SUCCESS_CLASS, T01);
+        ListStudentCommand command = new ListStudentCommand(INDEX_ONE, null);
+        CommandResult result = command.execute(new ModelStubAcceptingIndex());
+        assertEquals(expectedMessage, result.getFeedbackToUser());
+    }
+
+    @Test
+    public void execute_indexInput_invalidIndexFailure() {
+        String expectedMessage = Messages.MESSAGE_INVALID_TUTORIAL_DISPLAYED_INDEX;
+        ListStudentCommand command = new ListStudentCommand(INDEX_TWO, null);
+        try {
+            command.execute(new ModelStubAcceptingIndex());
+        } catch (CommandException ce) {
+            assertEquals(expectedMessage, ce.getMessage());
+        }
+    }
+
+    @Test
+    public void execute_indexInput_displayListMismatchFailure() {
+        String expectedMessage = Messages.MESSAGE_INDEX_LIST_MISMATCH + ListStudentCommand.MESSAGE_INDEX_USAGE;
+        ListStudentCommand command = new ListStudentCommand(INDEX_ONE, null);
+        try {
+            ModelStubAcceptingIndex model = new ModelStubAcceptingIndex();
```

Comment:
same as above, should we use assertThrows instread?

Reply:
Thanks! Will make the changes